Exception of type `System.Collections.Generic.KeyNotFoundException

```
$ openra 
Platform is Linux
Using SDL 2 with OpenGL renderer
Desktop resolution: 1920x1080
No custom resolution provided, using desktop resolution
Using resolution: 1920x1080
Detected OpenGL version: 4.5
Using OpenAL sound engine
Using default device
Available mods:
	cnc: Tiberian Dawn (playtest-20150408)
	d2k: Dune 2000 (playtest-20150408)
	modchooser: Mod Chooser (playtest-20150408)
	ra: Red Alert (playtest-20150408)
Loading mod: ra
Exception of type `System.Collections.Generic.KeyNotFoundException`: The given key was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[System.String,OpenRA.Mods.Common.Widgets.Logic.LobbyCountry].get_Item (System.String key) [0x00000] in <filename unknown>:0 
  at OpenRA.Mods.Common.Widgets.Logic.LobbyUtils.SetupEditableFactionWidget (OpenRA.Widgets.Widget parent, OpenRA.Network.Slot s, OpenRA.Network.Client c, OpenRA.Network.OrderManager orderManager, System.Collections.Generic.Dictionary`2 countries) [0x00000] in <filename unknown>:0 
  at OpenRA.Mods.Common.Widgets.Logic.LobbyLogic.UpdatePlayerList () [0x00000] in <filename unknown>:0 
  at (wrapper delegate-invoke) <Module>:invoke_void__this__ ()
  at (wrapper delegate-invoke) <Module>:invoke_void__this__ ()
  at OpenRA.Game.SyncLobbyInfo () [0x00000] in <filename unknown>:0 
  at OpenRA.Network.UnitOrders.ProcessOrder (OpenRA.Network.OrderManager orderManager, OpenRA.World world, Int32 clientId, OpenRA.Order order) [0x00000] in <filename unknown>:0 
  at OpenRA.Network.OrderManager.TickImmediate () [0x00000] in <filename unknown>:0 
  at OpenRA.Sync+<CheckSyncUnchanged>c__AnonStorey0.<>m__0 () [0x00000] in <filename unknown>:0 
  at OpenRA.Sync.CheckSyncUnchanged[Boolean] (OpenRA.World world, System.Func`1 fn) [0x00000] in <filename unknown>:0 
  at OpenRA.Sync.CheckSyncUnchanged (OpenRA.World world, System.Action fn) [0x00000] in <filename unknown>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x00000] in <filename unknown>:0 
  at OpenRA.Game.LogicTick () [0x00000] in <filename unknown>:0 
  at OpenRA.Game.Loop () [0x00000] in <filename unknown>:0 
  at OpenRA.Game.Run () [0x00000] in <filename unknown>:0 
  at OpenRA.Program.Run (System.String[] args) [0x00000] in <filename unknown>:0 
  at OpenRA.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0 
```

I receive this error when i try to open a new skirmish game

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/10956475-exception-of-type-system-collections-generic-keynotfoundexception?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

```
$ mono OpenRA.Utility.exe ra --map-hash /home/user/.openra/maps/
Error: Utility application crashed. See utility.log for details

Unhandled Exception:
System.InvalidOperationException: Required file map.yaml not present in this map
  at OpenRA.Map.AssertExists (System.String filename) [0x00000] in <filename unknown>:0 
  at OpenRA.Map..ctor (System.String path) [0x00000] in <filename unknown>:0 
  at OpenRA.Mods.Common.UtilityCommands.GetMapHashCommand.Run (OpenRA.ModData modData, System.String[] args) [0x00000] in <filename unknown>:0 
  at OpenRA.Utility.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: Required file map.yaml not present in this map
  at OpenRA.Map.AssertExists (System.String filename) [0x00000] in <filename unknown>:0 
  at OpenRA.Map..ctor (System.String path) [0x00000] in <filename unknown>:0 
  at OpenRA.Mods.Common.UtilityCommands.GetMapHashCommand.Run (OpenRA.ModData modData, System.String[] args) [0x00000] in <filename unknown>:0 
  at OpenRA.Utility.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0 
```

Anyway, i solved the problem removing the hash from settings.yaml. Thanks guys

--- a/OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs
+++ b/OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs
@@ -144,8 +144,8 @@ namespace OpenRA.Mods.Common.Widgets.Logic
 			colorPreview = lobby.Get<ColorPreviewManagerWidget>("COLOR_MANAGER");
 			colorPreview.Color = Game.Settings.Player.Color;
 
-			foreach (var c in modRules.Actors["world"].Traits.WithInterface<CountryInfo>().Where(c => c.Selectable))
-				countries.Add(c.Race, new LobbyCountry { Name = c.Name, Side = c.Side, Description = c.Description });
+			foreach (var c in modRules.Actors["world"].Traits.WithInterface<CountryInfo>())
+				countries.Add(c.Race, new LobbyCountry { Selectable = c.Selectable, Name = c.Name, Side = c.Side, Description = c.Description });
 
 			var gameStarting = false;
 			Func<bool> configurationDisabled = () => !Game.IsHost || gameStarting ||
@@ -829,6 +829,7 @@ namespace OpenRA.Mods.Common.Widgets.Logic
 
 	public class LobbyCountry
 	{
+		public bool Selectable;
 		public string Name;
 		public string Description;
 		public string Side;

--- a/OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs
+++ b/OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs
@@ -121,7 +121,8 @@ namespace OpenRA.Mods.Common.Widgets.Logic
 				return item;
 			};
 
-			var options = countries.GroupBy(c => c.Value.Side).ToDictionary(g => g.Key ?? "", g => g.Select(c => c.Key));
+			var options = countries.Where(c => c.Value.Selectable).GroupBy(c => c.Value.Side)
+				.ToDictionary(g => g.Key ?? "", g => g.Select(c => c.Key));
 
 			dropdown.ShowDropDown("RACE_DROPDOWN_TEMPLATE", 150, options, setupItem);
 		}

